lib/cmd/fh3/undeploy.js

```diff
@@ -0,0 +1,50 @@
+/* globals i18n */
+module.exports = undeploy;
+undeploy.undeploy = undeploy;
+
+undeploy.desc = i18n._("Undeploy a cloud app");
+undeploy.usage = "fhc undeploy <app-id> [apptype] --env=<environment>";
+
+var log = require("../../utils/log");
+var fhc = require("../../fhc");
+var fhreq = require("../../utils/request");
+var common = require("../../common");
+var ini = require('../../utils/ini');
+
+// Main undeploy entry point
+function undeploy(argv, cb) {
+  var args = argv._;
+  if (args.length === 0) {
+    return cb(undeploy.usage);
+  }
+
+  var appId = fhc.appId(args[0]);
+  var appType = args[1];
+  var deployTarget = ini.getEnvironment(argv);
+
+  return undeployApp(appId, deployTarget, appType, cb);
+}
+
+function undeployApp(appId, deployTarget, appType, cb) {
+  var type = appType || "cloud";
+  var params = {
+    guid: appId,
+    apptype: type,
+    deploytarget: deployTarget
+  };
+
+  common.doApiCall(fhreq.getFeedHenryUrl(), "box/srv/1.1/environments/" + deployTarget + "/apps/" + appId + "/undeploy", params, i18n._("Error undeploying app: "), function (err, data) {
```

Comment:
I'm not sure that deploy operation is supported for new version of feedhenry/on-prem which can lead to some unexpected results. We will possibly need to verify that with on-prem version as I see that we use old API here.